ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx

```diff
@@ -30,26 +30,25 @@ export const DashboardToolbar = (props: DashboardToolbarProps) => {
   const { dashboardName } = props;
 
   const { isEditMode, setEditMode } = useEditMode();
-  const { openAddPanelGroup, openAddPanel, save } = useDashboardActions();
+  const { openAddPanelGroup, openAddPanel } = useDashboardActions();
   const isLaptopSize = useMediaQuery(useTheme().breakpoints.up('sm'));
-  const [originalSpec, setOriginalSpec] = useState<DashboardSpec | undefined>(undefined);
-  const { spec, resetSpec } = useDashboardSpec();
+  const [originalDashboard, setOriginalDashboard] = useState<DashboardResource | undefined>(undefined);
+  const { dashboardResource, resetDashboardResource } = useDashboardResource();
 
   const onEditButtonClick = () => {
-    setOriginalSpec(spec);
+    setOriginalDashboard(dashboardResource);
     setEditMode(true);
   };
 
   const onCancelButtonClick = () => {
     // Reset to the original spec and exit edit mode
-    if (originalSpec) {
-      resetSpec(originalSpec);
+    if (originalDashboard) {
+      resetDashboardResource(originalDashboard);
     }
     setEditMode(false);
   };
 
   const onSave = () => {
-    save();
```

Comment:
I assume the state for panels / panel groups is still being set at some point in the editing process, so it's OK we don't call `save` here?

Reply:
yes every time you click the edit button, the state is saved in `originalDashboard`